src/api/registerForEvents.js

```diff
@@ -19,14 +20,39 @@ type RegisterForEventsParams = {|
   |},
 |};
 
+export const transformUserOrBot = <T: User | CrossRealmBot>(
+  rawUserOrBot: $FlowFixMe, // server data pre-transformation
```

Comment:
These fixmes can be avoided: here, write `{| ...T, avatar_url: string |}`, and then on `transform` spell out the same thing for each of `realm_users` and its two friends.

Better yet, I think: do that, and pull out the type used for `transform`'s argument, giving it a name like `RawInitialData`.